aptos-move/framework/aptos-stdlib/sources/cryptography/groups.move

```diff
@@ -0,0 +1,749 @@
+        abort_if_feature_disabled();
+        Scalar<G> {
+            handle: scalar_from_u64_internal<G>(value)
+        }
+    }
+
+    /// Compute `-x` for scalar `x`.
+    public fun scalar_neg<G>(x: &Scalar<G>): Scalar<G> {
+        abort_if_feature_disabled();
+        Scalar<G> {
+            handle: scalar_neg_internal<G>(x.handle)
+        }
+    }
+
+    /// Compute `x + y` for scalar `x` and `y`.
+    public fun scalar_add<G>(x: &Scalar<G>, y: &Scalar<G>): Scalar<G> {
+        abort_if_feature_disabled();
+        Scalar<G> {
+            handle: scalar_add_internal<G>(x.handle, y.handle)
+        }
+    }
+
+    /// Compute `x * y` for scalar `x` and `y`.
+    public fun scalar_mul<G>(x: &Scalar<G>, y: &Scalar<G>): Scalar<G> {
+        abort_if_feature_disabled();
+        Scalar<G> {
+            handle: scalar_mul_internal<G>(x.handle, y.handle)
+        }
+    }
+
+    /// Compute `x^(-1)` for scalar `x`, if defined.
+    public fun scalar_inv<G>(x: &Scalar<G>): Option<Scalar<G>> {
+        abort_if_feature_disabled();
+        let (succeeded, handle) = scalar_inv_internal<G>(x.handle);
+        if (succeeded) {
+            let scalar = Scalar<G> { handle };
+            std::option::some(scalar)
+        } else {
+            std::option::none()
+        }
+    }
+
+    /// Check if `x == y` for scalar `x` and `y`.
+    public fun scalar_eq<G>(x: &Scalar<G>, y: &Scalar<G>): bool {
+        abort_if_feature_disabled();
+        scalar_eq_internal<G>(x.handle, y.handle)
+    }
+
+    /// Get the identity of group `G`.
+    public fun group_identity<G>(): Element<G> {
+        abort_if_feature_disabled();
+        Element<G> {
+            handle: group_identity_internal<G>()
+        }
+    }
+
+    /// Get the generator of group `G`.
+    public fun group_generator<G>(): Element<G> {
+        abort_if_feature_disabled();
+        Element<G> {
+            handle: group_generator_internal<G>()
+        }
+    }
+
+    /// Compute `-P` for group element `P`.
+    public fun element_neg<G>(element_p: &Element<G>): Element<G> {
+        abort_if_feature_disabled();
+        Element<G> {
+            handle: element_neg_internal<G>(element_p.handle)
+        }
+    }
+
+    /// Compute `P + Q` for group element `P` and `Q`.
+    public fun element_add<G>(element_p: &Element<G>, element_q: &Element<G>): Element<G> {
+        abort_if_feature_disabled();
+        Element<G> {
+            handle: element_add_internal<G>(element_p.handle, element_q.handle)
+        }
+    }
+
+    /// Compute `2P` for group element `P`.
+    public fun element_double<G>(element_p: &Element<G>): Element<G> {
+        abort_if_feature_disabled();
+        Element<G> {
+            handle: element_double_internal<G>(element_p.handle)
+        }
+    }
+
+    /// Compute `k*P` for scalar `k` and group element `P`.
+    public fun element_scalar_mul<G>(scalar_k: &Scalar<G>, element_p: &Element<G>): Element<G> {
```

Comment:
Can you add a `generator_scalar_mul(a: &Scalar<F>)` function. This function computes `a * G` where `G` is the generator.

We want this separate function because some libraries offer faster scalar multiplications w.r.t. the generator as the base. For example, on Ristretto255:

```
ristretto255/basepoint_mul  time:   [10.259 µs 10.263 µs 10.267 µs]

ristretto255/point_mul  time:   [40.163 µs 40.187 µs 40.212 µs]
```

If your library has this too, then in the native, you can use the faster library call. Otherwise, just fetch the generator in the native and do a normal scalar multiplication.